etna/reconciliation/base.py

```diff
@@ -0,0 +1,101 @@
+from abc import ABC
+from abc import abstractmethod
+from typing import Optional
+
+import pandas as pd
+from scipy.sparse import csr_matrix
+
+from etna.core import BaseMixin
+from etna.datasets import TSDataset
+
+
+class BaseReconciliator(ABC, BaseMixin):
+    """Base class to hold reconciliation methods."""
+
+    def __init__(self, target_level: str, source_level: str):
+        """Init BaseReconciliator.
+
+        Parameters
+        ----------
+        target_level:
+            Level to be reconciled from the forecasts.
+        source_level:
+            Level to be forecasted.
+        """
+        self.target_level = target_level
+        self.source_level = source_level
+        self.mapping_matrix: Optional[csr_matrix] = None
+
+    @abstractmethod
+    def fit(self, ts: TSDataset) -> "BaseReconciliator":
+        """Fit the reconciliator parameters.
+
+        Parameters
+        ----------
+        ts:
+            TSDataset on the level which is lower or equal to ``target_level``, ``source_level``.
+
+        Returns
+        -------
+        :
+            Fitted instance of reconciliator.
+        """
+        pass
+
+    def aggregate(self, ts: TSDataset) -> TSDataset:
+        """Aggregate the dataset to the ``source_level``.
+
+        Parameters
+        ----------
+        ts:
+            TSDataset on the level which is lower or equal to ``source_level``.
+
+        Returns
+        -------
+        :
+            TSDataset on the ``source_level``.
+        """
+        ts_aggregated = ts.get_level_dataset(target_level=self.source_level)
+        return ts_aggregated
+
+    def reconcile(self, ts: TSDataset) -> TSDataset:
```

Comment:
I think here we also need to check if the target level is higher than that source.

Reply:
It is not required here as this method is common for both top-down and bottom-up